src/main/scala/loamstream/apps/AppWiring.scala

```diff
@@ -132,34 +133,34 @@ object AppWiring extends Loggable {
   }
   
   def jobFilterForDryRun(intent: Intent.DryRun, makeDao: => LoamDao): JobFilter = {
-    val (jobFilter, _) = makeJobFilterAndExecutionRecorder(intent.jobFilterIntent, intent.hashingStrategy, makeDao)
-    
-    jobFilter
+    AppWiring.makeJobFilter(intent.jobFilterIntent, intent.hashingStrategy, makeDao)
   }
   
   def forRealRun(intent: Intent.RealRun, makeDao: => LoamDao): AppWiring = {
     new DefaultAppWiring(intent, makeDao = makeDao)
   }
 
-  private[AppWiring] def makeJobFilterAndExecutionRecorder(
+  private[AppWiring] def makeJobFilter(
       jobFilterIntent: JobFilterIntent,
       hashingStrategy: HashingStrategy,
-      getDao: => LoamDao): (JobFilter, ExecutionRecorder) = {
+      getDao: => LoamDao): JobFilter = {
     
     val dao = getDao
     
     import JobFilterIntent._
     
-    val jobFilter = jobFilterIntent match {
+    jobFilterIntent match {
```

Comment:
This may be totally out of context, so ignore if it is...

I have something similar here (filtering jobs) in the aggregator, which was originally done via regular expressions. This ended up being a real PITA not only for myself, but for Marcin as well. I found it was much better to end up using globs, and so I wrapped Java path globs:

https://github.com/broadinstitute/dig-aggregator-core/blob/master/src/main/scala/org/broadinstitute/dig/aggregator/core/Glob.scala

It ended up being a lot nicer being able to reference a job/input like `"T2D*"` instead of `"T2D.*"`. In case you'd like to end up doing the same thing and just copy the code over?

Reply:
Also, here's an example of those globs in use:

https://github.com/broadinstitute/dig-aggregator-core/blob/e88bff521203283d80871dae6d4566cb939cfe18/src/main/scala/org/broadinstitute/dig/aggregator/core/processors/DatasetProcessor.scala#L38

Reply:
Thanks for this.  I noticed that code in the Aggregator PR, and was interested to learn about `PathMatcher`s.  (Despite it being a zillion years old, there's lots of good stuff in `java.nio.file` that I'm only just finding out about.)

In the LS case, what has to get matched are user-defined strings which can be anything, not `Path`s.  I'll see if Ryan even uses the CLI flag in question.